pex/bin/pex.py

```diff
@@ -302,6 +302,18 @@ def configure_clp_pex_options(parser):
            'likely will not be reproducible, meaning that if you were to run `./pex -o` with the '
            'same inputs then the new pex would not be byte-for-byte identical to the original.')
 
+  group.add_option(
+      '--use-system-time', '--no-use-system-time',
```

Comment:
Open to suggestions for a better name.

The main use case I want to optimize for is when deterministic time will be the default. For example, atm `--use-deterministic-time` would probably be a good name, but once we make the switch then `--no-use-deterministic-time` wouldn't be very descriptive for what the behavior would end up being.

Reply:
In https://github.com/pantsbuild/pants/pull/6094 we made a flag `-jar-creation-time` which, if not set, defaulted to the current time. Not sure whether this is a good or bad default for pex, but it's some prior art :)

Reply:
@illicitonion : I think that we agreed on the other issue that we want this enabled by default, so the default would need to be stable.

@Eric-Arellano : This name looks good I think?